compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/expression/FirGetClassCallChecker.kt

```diff
@@ -36,10 +35,20 @@ object FirGetClassCallChecker : FirBasicExpressionChecker() {
         //
         // Only the 2nd example is valid, and we want to check if token type QUEST doesn't exist at the same level as COLONCOLON.
         val markedNullable = source.getChild(QUEST, depth = 1) != null
-        if (argument.isNullableLHSForCallableReference || markedNullable) {
-            reporter.reportOn(source, FirErrors.NULLABLE_TYPE_IN_CLASS_LITERAL_LHS, context)
+        val argument = expression.argument
+        val isNullable = markedNullable ||
+                (argument as? FirResolvedQualifier)?.isNullableLHSForCallableReference == true ||
+                argument.typeRef.coneType.isMarkedNullable
+        if (isNullable) {
```

Comment:
Thanks. I noticed similar examples, but didn't remember why I assumed it would require more learning or something. Anyway, yeah, unless `T` is reified or has a subtype constraint with non-null type, we should report nullable expression of LHS of class literals. Alas(?), checking a subtype of `Any` is added as part of `KClass` signature, so it's a bit hard to figure out commits' order and/or merging. Hence, added as the last commit.